HierarchicalDataConverter/Catalyst.HierarchicalDataConverter.AutomatedTests/GrandparentParentChildTest.cs

```diff
@@ -0,0 +1,441 @@
+                                                       },
+                                                   new ObjectAttributeValue
+                                                       {
+                                                           AttributeName = "ParentKeyFields",
+                                                           AttributeValue = "['id1']"
+                                                       },
+                                                   new ObjectAttributeValue
+                                                       {
+                                                           AttributeName = "ChildKeyFields",
+                                                           AttributeValue = "['id1']"
+                                                       },
+                                                   new ObjectAttributeValue
+                                                       {
+                                                           AttributeName = "GenerationGap",
+                                                           AttributeValue = "1"
+                                                       }
+                                               }
+                                       }
+                               }
+            };
+        }
+
+        private Binding GetNestedBindingLevel2Source()
+        {
+            var destinationEntity = this.GetNestedDestinationEntity();
+            var level1SourceEntity = this.GetLevel1SourceEntity();
+            var level2SourceEntity = this.GetLevel2SourceEntity();
+            return new Binding
+                       {
+                           DestinationEntityId = destinationEntity.Id,
+                           ContentId = Guid.NewGuid(),
+                           DataMartId = this.DataMartId,
+                           Classification = "Generic",
+                           Name = "NestedBindingLevel2Source",
+                           SourceConnectionId = this.SourceConnectionId,
+                           BindingType = "Nested",
+                           Status = "Active",
+                           LoadTypeCode = "Full",
+                           Id = 2,
+                           SourcedByEntities =
+                               {
+                                   new SourceEntityReference
+                                       {
+                                           SourceEntityId = level2SourceEntity.Id
+                                       }
+                               }
+            };
+        }
+
+        private Entity GetNestedDestinationEntity()
+        {
+            return new Entity
+                       {
+                           Id = 6357,
+                           DataMartId = this.DataMartId,
+                           Fields =
+                               {
+                                   new Field
```

Comment:
This one also needs to have 'Level0Entity__id0', 'Level1Entity__id0', 'Level2Entity__id0', 'Level1Entity__id1', 'Level2Entity__id1' -- according the parent/child key field designations above.